src/autoscaler/metricscollector/cmd/metricscollector/metricscollector_test.go

```diff
@@ -284,4 +285,21 @@ var _ = Describe("MetricsCollector", func() {
 			rsp.Body.Close()
 		})
 	})
+
+	Context("when using streaming for metrics collection", func() {
```

Comment:
We can add a describe for REST API test, and then have two contexts for polling and streaming